### DISCUSSION TITLE
OCR依赖库

### DISCUSSION THREAD

**Discussion by emako** [babalae/better-genshin-impact#403] [Ideas]
试试https://gitee.com/raoyutian/PaddleOCRSharp

https://github.com/ZGGSONG/STranslate/issues/75

**Comment by huiyadanli** [comment]
跟我现在使用的是一样的依赖，表现应该没有差别，但是看了下代码水准差别还是有点大的。

**Reply by emako** [reply]
他说这个快，不清楚

**Comment by huiyadanli** [comment]
上次没看另外一个issue，今天看了下

体积问题是因为用的推理平台不一样
![image](https://github.com/babalae/better-genshin-impact/assets/15783049/e36b0267-9321-43b6-9e08-ae22bb7ff719)

使用的 [Mkldnn](https://github.com/oneapi-src/oneDNN) ， issue内提到 PaddleSharp 慢应该也是因为用了不同的推理平台。
https://github.com/sdcb/PaddleSharp/blob/master/README_CN.md#paddle-%E8%AE%BE%E5%A4%87

具体非常慢的问题可以参考 https://github.com/sdcb/PaddleSharp/issues/75

我直接使用的ONNX，体积稍微小一点，速度应该更加快。
追求速度还可以内置2G的CUDA，然后走GPU推理

如果我使用 https://gitee.com/raoyutian/PaddleOCRSharp 意味者我要新增一个推理平台和100M+的类库。（关键这个PaddleOCRSharp也没啥亮点，就是API调用。。。图像甚至转成byte后再处理，很难想象这个库会比PaddleSharp快）
BGI后面的方向应该是剥离Paddle的C++库，直接使用ONNX推理转换后的3大类模型，进一步提速与减小体积
